### PR TITLE
Update Sphinx theme

### PR DESCRIPTION
**Pull request by arnaucasau**
### Summary
Updates to new Ecosystem theme.

For more information: https://github.com/Qiskit/qiskit_sphinx_theme/issues/232

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -42,6 +42,7 @@
               'reno.sphinxext',
               'sphinx.ext.intersphinx',
               "sphinxcontrib.jquery",  # Remove when changing html_theme to qiskit_ecosystem.
+              'qiskit_sphinx_theme'
```
Comment (Eric-Arellano) [inline docs/conf.py:45]:
```suggestion
              'qiskit_sphinx_theme',
```

File: docs/conf.py [hunk 2]
```diff
@@ -42,6 +42,7 @@
               'reno.sphinxext',
               'sphinx.ext.intersphinx',
               "sphinxcontrib.jquery",  # Remove when changing html_theme to qiskit_ecosystem.
```
Comment (Eric-Arellano) [inline docs/conf.py:44]: Can delete this now

**Review by Eric-Arellano — APPROVED** [review]

**Review by mtreinish — APPROVED** [review]

**Comment by mtreinish** [comment]
This LGTM, but CI is failing because of a dependency conflict on the sphinx version: https://github.com/Qiskit/qiskit-neko/actions/runs/6354846073/job/17264430483?pr=32#step:7:108 I think once that is sorted and you mark this as ready for review we should be good to merge this.

**Comment by arnaucasau** [comment]
@mtreinish Thank you! Yes, the conflict is with the Sphinx version, and that version needs at leat python 3.8 as well. I will update the version of Python in another pull request to work with python 3.8 - 3.11.

**Review by mtreinish — APPROVED** [review]

**Pull request merged**